learning_journal/views.py

```diff
@@ -36,20 +36,29 @@ def view(request):
         return HTTPNotFound()
     return {'entry': entry}
 
-@view_config(route_name='action', match_param='action=create', renderer='templates/edit.jinja2')
+@view_config(route_name='create', renderer='templates/edit.jinja2')
 def create(request):
     entry = Entry()
-    form = EntryCreateForm(request.POST)
+    form = EntryCreateForm(request.POST) # adjust row and column values here? see form widgets
     if request.method == 'POST' and form.validate():
         form.populate_obj(entry)
         DBSession.add(entry)
         return HTTPFound(location=request.route_url('home'))  # http redirect to home page
-    return {'form': form, 'action': request.matchdict.get('action')}
+    return {'form': form} # erased action
 
-@view_config(route_name='action', match_param='action=edit', renderer='string')
+@view_config(route_name='edit', renderer='templates/edit.jinja2')
 def update(request):
-    return 'edit page'
-
+    this_id = request.matchdict.get('id', -1)
+    entry = Entry.by_id(this_id)
+    if not entry:
+        return HTTPNotFound()
+    form = EntryEditForm(request.POST, entry)
+    if request.method == 'POST' and form.validate():
+        form.populate_obj(entry)
+        DBSession.add(entry)
```

Comment:
this line of code is not needed.  The `entry` object you have is a result of making a query from the DBSession.  That means that the session is actually already aware of the object.  If your call to `form.populate_object()` resulted in any change to the entry, then DBSession already knows it is changed (or `dirty`) and will have a record that it needs to be updated on session commit.